pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -74,39 +100,38 @@ func NewNodeController(
 		registerRetryCount: registerRetryCount,
 		podEvictionTimeout: podEvictionTimeout,
 		lookupIP:           net.LookupIP,
+		now:                util.Now,
 	}
 }
 
-// Run creates initial node list and start syncing instances from cloudprovider if any.
-// It also starts syncing cluster node status.
+// Run creates initial node list and start syncing instances from cloudprovider, if any.
+// It also starts syncing or monitoring cluster node status.
 // 1. RegisterNodes() is called only once to register all initial nodes (from cloudprovider
 //    or from command line flag). To make cluster bootstrap faster, node controller populates
 //    node addresses.
-// 2. SyncCloud() is called periodically (if enabled) to sync instances from cloudprovider.
+// 2. SyncCloudNodes() is called periodically (if enabled) to sync instances from cloudprovider.
 //    Node created here will only have specs.
-// 3. SyncNodeStatus() is called periodically (if enabled) to sync node status for nodes in
-//    k8s cluster.
+// 3. Depending on how k8s is configured, there are two ways of syncing the node status:
+//   3.1 SyncProbedNodeStatus() is called periodically to sync node status for nodes in k8s cluster.
```

Comment:
3.1 and 3.2 are confusing if the person isn't already familiar with the code. I would rewrite as
3.1 SyncProbedNodeStatus() is called periodically to trigger master to probe kubelet, and incorporate the resulting node status
3.2 MonitorNodeStatus() is called periodically to incorporate the results of node status pushed from kubelet to master